src/components/CurrentWalletAdvanced.tsx

```diff
@@ -1,49 +1,69 @@
 import React, { useState, useEffect } from 'react'
 import * as rb from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
+// @ts-ignore
 import DisplayAccounts from './DisplayAccounts'
+// @ts-ignore
 import DisplayAccountUTXOs from './DisplayAccountUTXOs'
+// @ts-ignore
 import DisplayUTXOs from './DisplayUTXOs'
+// @ts-ignore
 import { useCurrentWallet, useCurrentWalletInfo, useSetCurrentWalletInfo } from '../context/WalletContext'
 import * as Api from '../libs/JmWalletApi'
 
+type Utxos = any[]
+type Alert = { message: string; variant: string }
+
 export default function CurrentWalletAdvanced() {
   const { t } = useTranslation()
   const currentWallet = useCurrentWallet()
   const walletInfo = useCurrentWalletInfo()
   const setWalletInfo = useSetCurrentWalletInfo()
-  const [fidelityBonds, setFidelityBonds] = useState(null)
-  const [utxos, setUtxos] = useState(null)
+  const [fidelityBonds, setFidelityBonds] = useState<Utxos | null>(null)
+  const [utxos, setUtxos] = useState<Utxos | null>(null)
   const [showUTXO, setShowUTXO] = useState(false)
-  const [alert, setAlert] = useState(null)
+  const [alert, setAlert] = useState<Alert | null>(null)
   const [isLoading, setIsLoading] = useState(true)
 
   useEffect(() => {
+    if (!currentWallet) {
+      setAlert({ variant: 'danger', message: t('current_wallet.error_loading_failed') })
+      setIsLoading(false)
+      return
+    }
+
     const abortCtrl = new AbortController()
-    const { name, token } = currentWallet
+    const { name: walletName, token } = currentWallet
 
-    const setUtxoData = (utxos) => {
+    const setUtxoData = (utxos: Utxos) => {
       setUtxos(utxos)
       setFidelityBonds(utxos.filter((utxo) => utxo.locktime))
     }
 
     setAlert(null)
     setIsLoading(true)
 
-    const loadingWallet = Api.getWalletDisplay({ walletName: name, token, signal: abortCtrl.signal })
+    const loadingWallet = Api.getWalletDisplay({ walletName, token, signal: abortCtrl.signal })
       .then((res) =>
-        res.ok ? res.json() : Promise.reject(new Error(res.message || t('current_wallet.error_loading_failed')))
+        res.ok
+          ? res.json()
+          : Api.Helper.toError(res, t('current_wallet.error_loading_failed')).then((e) => {
```

Comment:
Would it work to throw the error inside `toError` (which would then need to be renamed to `throwError` or something like that)? Then we could get rid of the nested `.then()` call which is a bit cumbersome to read/write imo.